### PR TITLE
Use mu_main to reduce boilerplate in unit tests

### PR DESCRIPTION
**Pull request by ret2libc**
* Disable colors on Windows because terminal looks bad otherwise
* Use environment on Windows to set PATH and make sure unit tests use
  the compiled libraries and not the installed ones
* Define ssize_t on Windows as well by using SSIZE_T
* A bit of refactoring in rz_file_mmap API
* Remove io_mmap plugin and just keep io_default, which does a mmap if possible or just directly access the file if not possible.
* Simplify io_default code by just using RzBuffer API
* Add unit test for io_default
* Add unit test for rz_file_mmap API
* Allow IO maps of size 0. This is useful when creating a new file (e.g. `rizin -w newfile`) and you use `r` to resize the file. By having a map of 0 size, the map will be resized as well. Without this, the newfile cannot be easily mmaped and viewed in `om`.

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**

I have marked test_rzpipe as broken on windows because it does not work for now and I want to fix that by using the improved RzSubprocess API. 

This is needed for https://github.com/rizinorg/rizin/pull/243 , so I can have unit tests to make sure stuff works well on linux & windows.

**Test plan**

AppVeyor should be green even with tests enabled

### CODE REVIEW

File: librz/magic/file.h [hunk 1]
```diff
@@ -46,9 +46,6 @@
 /* Do this here and now, because struct stat gets re-defined on solaris */
 #include <sys/stat.h>
 #include <stdarg.h>
-#ifdef _MSC_VER
```
Comment (XVilka) [inline librz/magic/file.h:49]: One more embedded file for https://github.com/rizinorg/rizin/issues/209

**Review by XVilka — APPROVED** [review]

File: librz/magic/file.h (hunk 1 above)
Comment (ret2libc) [reply librz/magic/file.h:49]: Yeah, however I am not sure how to use this if not embedding. AFAIR this is slightly different from the linux `magic` and I don't think distros have a `magic-bsd` package.

Comment (XVilka) [reply librz/magic/file.h:49]: Yes, embedding is fine. We just should note imported version somewhere and our changes.

**Comment by XVilka** [comment]
See also the changed error message:
```
[XX] C:\projects\rizin\test\db\tools\rz_find rz-find can't open
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rz-find -n  -f 0x00010000 -t 0x00020000 -r -X -s 250382 incorrect_file | grep incorrect_file' -
-- stderr
@@ -1,2 +1,2 @@
-rz_sys_open: No such file or directory
+rz_sys_open: (0x2) The system cannot find the file specified.
 Cannot open file 'incorrect_file'
```
https://ci.appveyor.com/project/rizinorg/rizin/builds/37056378/job/5pmtiku169x60fi5#L4020

**Review by thestr4ng3r — COMMENTED** [review]
Changes lgtm but the history should be cleaned up if possible to split into `mu_main`, `mmap` removal, windows-specific stuff...

**Comment by ret2libc** [comment]
> Changes lgtm but the history should be cleaned up if possible to split into `mu_main`, `mmap` removal, windows-specific stuff...

Done. WDYT now?

File: test/unit/test_buf.c [hunk 2]
```diff
@@ -154,27 +154,54 @@ bool test_r_buf_mmap(void) {
 	write (fd, content, length);
 	close (fd);
 
-	b = rz_buf_new_mmap (filename, RZ_PERM_RW);
+	b = rz_buf_new_mmap (filename, O_RDWR, 0);
```
Comment (yossizap) [inline test/unit/test_buf.c:157]: Might be out of scope for this PR but I think that `rz_sys_open` should accept `RZ_PERM` and use your `rizinperms2open` in the same manner as `rz_io_def_mmap_create_new_file` to keep things consistent. 
Especially when `O_CREAT` is implicitly toggled in `RZ_PERMS` for WR while having to set it explicitly with functions that accept `O_*` perms.

**Review by yossizap — APPROVED** [review]

File: test/unit/test_buf.c (hunk 2 above)
Comment (ret2libc) [reply test/unit/test_buf.c:157]: I agree. I started doing that but it required many changes all over the place so I ended up keeping it like this for now.

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**